lib/Echidna/Internal/Runner.hs

```diff
@@ -565,24 +565,33 @@ takeLines sloc =
   snd . Map.split (spanStartLine sloc - 1) .
   declarationSource
 
+
+checkFilePath :: FilePath -> IO Bool
+checkFilePath path = do
+   strOrExc <- try $ readFile path
+   case (strOrExc :: Either IOError String) of
+     Left  _        -> return False
+     Right _        -> return True
+
 readDeclaration :: MonadIO m => Span -> m (Maybe (Declaration ()))
 readDeclaration sloc =
   runMaybeT $ do
     path <- liftIO . makeRelativeToCurrentDirectory $ spanFile sloc
-
-    (name, Pos (Position _ line0 _) src) <- MaybeT $
-      Discovery.readDeclaration path (spanEndLine sloc)
+    exists <- liftIO $ checkFilePath path
+    (name, Pos (Position _ line0 _) src) <- MaybeT $ Discovery.readDeclaration path (spanEndLine sloc)
 
     let
       line =
         fromIntegral line0
-
-    pure . Declaration path line name .
-      Map.fromList .
-      zip [line..] .
-      zipWith (Line ()) [line..] $
-      lines src
-
+    if (not exists)
```

Comment:
I don't think the parens in this conditional are necessary (but also idk quite what to do about style in this file, so feel free to leave them if you think they're more consistent or etc.)